### PR TITLE
MINOR: Add 3.7 upgrade notes

### PR DESCRIPTION
**Pull request by stanislavkozlovski**
This patch adds the 3.7 upgrade notes.

### CODE REVIEW

File: docs/upgrade.html [hunk 1]
```diff
@@ -19,24 +19,94 @@
 
 <script id="upgrade-template" type="text/x-handlebars-template">
 
-<h4><a id="upgrade_3_7_0" href="#upgrade_3_7_0">Upgrading to 3.7.0 from any version 0.8.x through 3.6.x</a></h4>
+    <h4><a id="upgrade_3_7_0" href="#upgrade_3_7_0">Upgrading to 3.7.0 from any version 0.8.x through 3.6.x</a></h4>
```
Comment (stanislavkozlovski) [inline docs/upgrade.html:22]: this wasn't aligned properly before

**Review by divijvaidya — APPROVED** [review]

**Review by mimaison — APPROVED** [review]
LGTM, I generated the docs and validated it render correctly.

**Comment by divijvaidya** [comment]
As an aside - we need a similar PR for ops.html as well since kafka-site is ahead of trunk here right now [1]. This doesn't block 3.7 and could be done later as well. Just have to make sure that we don't overwrite the 37 section in kafka-site when we copy generated docs from here.

[1] https://github.com/apache/kafka-site/pull/578/files#diff-592961a45575efec39dbcc010a6e0c7f40c81a0afa61ad0c7d4a8fe13739f44a

**Comment by stanislavkozlovski** [comment]
rebased with master. no conflicts. merging...

**Comment by stanislavkozlovski** [comment]
cherry-picked to 3.7 with 9603d4c9ef

**Pull request merged**